File: Python/Program/Motorcycle.py
from Vehicle import Vehicle

class Motorcycle(Vehicle):
    def __init__(self, plat_nomor, merk, tahun_produksi, warna, jenis_motor, kapasitas_tangki):
        super().__init__(plat_nomor, merk, tahun_produksi, warna)

        self.jenis_motor = jenis_motor
        self.kapasitas_tangki = kapasitas_tangki

    def get_jenis_motor(self):
        return self.jenis_motor

    def set_jenis_motor(self, jenis_motor):
        self.jenis_motor = jenis_motor

    def get_kapasitas_tangki(self):
        return self.kapasitas_tangki

    def set_kapasitas_tangki(self, kapasitas_tangki):
        self.kapasitas_tangki = kapasitas_tangki

File: CPP/Program/Motorcycle.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>
#include "Vehicle.cpp"

// Deklarasi namespace
using namespace std;

class Motorcycle : public Vehicle {

private:
    string jenis_motor;
    double kapasitas_tangki;

public:
    Motorcycle(string plat_nomor, string merk, int tahun_produksi, string warna, string jenis_motor, double kapasitas_tangki)
        : Vehicle(plat_nomor, merk, tahun_produksi, warna), jenis_motor(jenis_motor), kapasitas_tangki(kapasitas_tangki) {}

    string getJenisMotor() const {
        return jenis_motor;
    }

    void setJenisMotor(const string& jenis_motor) {
        this->jenis_motor = jenis_motor;
    }

    double getKapasitasTangki() const {
        return kapasitas_tangki;
    }

    void setKapasitasTangki(double kapasitas_tangki) {
        this->kapasitas_tangki = kapasitas_tangki;
    }
    
    // Destructor
    ~Motorcycle()
    {


    }
};